diff --git a/conf.py b/conf.py
--- a/conf.py
+++ b/conf.py
@@ -1 +1,10 @@
 autodoc_typehints = "none"
+typehints_use_rtype = False
+
+def typehints_formatter(annotation, config):
+    if (
+        getattr(annotation, "__module__", None) in {"_thread", "threading"}
+        and getattr(annotation, "__qualname__", None) in {"_local", "local"}
+    ):
+        return ":class:`~threading.local`"
+    return None
